### PR TITLE
sql, toc: refine character set and collation document

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

+ Address comments in https://github.com/pingcap/docs-cn/pull/2606.
+ Unify wording: `排序规则` and `collation` --> `排序规则`
+ Other minor format edits.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: reference/sql/characterset-and-collation.md [hunk 1]
```diff
@@ -306,17 +301,17 @@ insert into t1 values ('a ');
 Query OK, 1 row affected # MySQL 中，由于补齐空格比较，报错 Duplicate entry 'a '。
 ```
 
-### 新框架下的 Collation 支持
+### 新框架下的排序规则支持
 
-TiDB 4.0 新增了完整的 Collation 支持框架，从语义上支持了 Collation，并新增了配置开关 `new_collation_enabled_on_first_boostrap`，在集群初次初始化时决定是否启用新 Collation 框架。在该配置开关打开之后初始化集群，可以通过 `mysql`.`tidb` 表中的 `new_collation_enabled` 变量确认新 Collation 是否启用：
+TiDB 4.0 新增了完整的排序规则支持框架，从语义上支持了排序规则，并新增了配置开关 `new_collation_enabled_on_first_boostrap`，在集群初次初始化时决定是否启用新排序规则框架。在该配置开关打开之后初始化集群，可以通过 `mysql`.`tidb` 表中的 `new_collation_enabled` 变量确认新排序规则是否启用：
```
Comment (yikeke) [inline reference/sql/characterset-and-collation.md:306]:
```suggestion
TiDB 4.0 新增了完整的排序规则支持框架，从语义上支持了排序规则，并新增了配置开关 `new_collation_enabled_on_first_boostrap`，在集群初次初始化时决定是否启用新排序规则框架。在该配置开关打开之后初始化集群，可以通过 `mysql`.`tidb` 表中的 `new_collation_enabled` 变量确认是否启用新排序规则框架：
```

File: reference/sql/characterset-and-collation.md [hunk 2]
```diff
@@ -346,35 +341,35 @@ ERROR 1062 (23000): Duplicate entry 'a ' for key 'PRIMARY'
...
++ 数字或者中间变量的 coercibility 值为 `5`。
++ `NULL` 或者由 `NULL` 派生出的表达式的 coercibility 值为 `6`。
 
-在推断 Collation 时，TiDB 优先使用 coercibility 值较低的表达式的 Collation（与 MySQL 一致）。如果 coercibility 值相同，则按以下优先级确定 Collation：
+在推断排序规则时，TiDB 优先使用 coercibility 值较低的表达式的排序规则（与 MySQL 一致）。如果 coercibility 值相同，则按以下优先级确定排序规则：
 
 binary > utf8mb4_bin > utf8mb4_general_ci > utf8_bin > utf8_general_ci > latin1_bin > ascii_bin
 
-如果两个子表达式的 Collation 不相同，而且表达式的 coercibility 值都为 `0` 时，TiDB无法推断 Collation 并报错。
+如果两个子表达式的排序规则不相同，而且表达式的 coercibility 值都为 `0` 时，TiDB无法推断排序规则并报错。
```
Comment (yikeke) [inline reference/sql/characterset-and-collation.md:360]:
```suggestion
如果两个子表达式的排序规则不相同，而且表达式的 coercibility 值都为 `0` 时，TiDB 无法推断排序规则并报错。
```

File: reference/sql/characterset-and-collation.md [hunk 3]
```diff
@@ -34,17 +31,17 @@ SHOW CHARACTER SET;
 
 > **注意：**
 >
-> - 每种字符集都对应一个默认的 Collation，并且仅有一个。
+> 每个字符集可能对应多个排序规则，但与之默认对应的排序规则有且仅有一个。
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply reference/sql/characterset-and-collation.md:34]: LGTM

File: reference/sql/characterset-and-collation.md [hunk 4]
```diff
@@ -203,7 +198,7 @@ col_name {ENUM | SET} (val_list)
 
 如果列的字符集和排序规则没有设置，那么表的字符集和排序规则就作为其默认值。
 
-## 字符串 Character Sets 和 Collation
+## 字符串的字符集和排序规则
 
 每一字符串字符文字有一个字符集和一个比较排序规则，在使用字符串时指，此选项可选，如下：
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply reference/sql/characterset-and-collation.md:203]: LGTM

File: reference/sql/characterset-and-collation.md [hunk 5]
```diff
@@ -221,13 +216,13 @@ SELECT _utf8mb4'string';
 SELECT _utf8mb4'string' COLLATE utf8mb4_general_ci;
 ```
 
-规则，如下：
+规则如下：
 
-* 规则1： 指定 CHARACTER SET charset_name 和 COLLATE collation_name，则直接使用 CHARACTER SET charset_name 和COLLATE collation_name。
-* 规则2:    指定 CHARACTER SET charset_name 且未指定 COLLATE collation_name，则使用 CHARACTER SET charset_name 和 CHARACTER SET charset_name 默认的排序比较规则。
-* 规则3:   CHARACTER SET charset_name 和 COLLATE collation_name 都未指定，则使用 character_set_connection 和 collation_connection 系统变量给出的字符集和比较排序规则。
+* 规则 1：指定 `CHARACTER SET charset_name` 和 `COLLATE collation_name`，则直接使用 `CHARACTER SET charset_name` 和 `COLLATE collation_name`。
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply reference/sql/characterset-and-collation.md:221]: LGTM

File: reference/sql/characterset-and-collation.md [hunk 6]
```diff
@@ -221,13 +216,13 @@ SELECT _utf8mb4'string';
...
+* 规则 2：指定 `CHARACTER SET charset_name` 且未指定 `COLLATE collation_name`，则使用 `CHARACTER SET charset_name` 和 `CHARACTER SET charset_name` 默认的排序比较规则。
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply reference/sql/characterset-and-collation.md:222]: 是的，应该统一。

File: reference/sql/characterset-and-collation.md [hunk 7]
```diff
@@ -265,15 +260,15 @@ SET character_set_results = charset_name;
...
 
-## Character Sets 和 Collation 通用选择规则
+## 字符集和排序规则通用选择规则
 
-* 规则1： 指定 CHARACTER SET charset_name 和 COLLATE collation_name，则直接使用 CHARACTER SET charset_name 和COLLATE collation_name。
-* 规则2:    指定 CHARACTER SET charset_name 且未指定 COLLATE collation_name，则使用 CHARACTER SET charset_name 和 CHARACTER SET charset_name 默认的排序比较规则。
-* 规则3:   CHARACTER SET charset_name 和 COLLATE collation_name 都未指定，则使用更高优化级给出的字符集和排序比较规则。
+* 规则 1：指定 `CHARACTER SET charset_name` 和 `COLLATE collation_name`，则直接使用 `CHARACTER SET charset_name` 和 `COLLATE collation_name`。
+* 规则 2：指定 `CHARACTER SET charset_name` 且未指定 `COLLATE collation_name`，则使用 `CHARACTER SET charset_name` 和 `CHARACTER SET charset_name` 默认的排序比较规则。
+* 规则 3：`CHARACTER SET charset_name` 和 `COLLATE collation_name` 都未指定，则使用更高优化级给出的字符集和排序比较规则。
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply reference/sql/characterset-and-collation.md:271]: LGTM

File: reference/sql/characterset-and-collation.md [hunk 8]
```diff
@@ -265,15 +260,15 @@ SET character_set_results = charset_name;
 SET collation_connection = @@collation_database;
 ```
 
-## 集群，服务器，数据库，表，列，字符串 Character Sets 和 Collation 优化级
+## 集群、服务器、数据库、表、列、字符串的字符集和排序规则的优化级别
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply reference/sql/characterset-and-collation.md:263]: LGTM

File: reference/sql/characterset-and-collation.md [hunk 9]
```diff
@@ -283,15 +278,15 @@ SET collation_connection = @@collation_database;
 
 ## 排序规则支持
 
-Collation 的语法支持和语义支持受到配置项 [`new_collation_enable`](/reference/configuration/tidb-server/configuration-file.md#new_collations_enabled_on_first_bootstrap) 的影响。这里语法支持和语义支持有所区别。语法支持是指 TiDB 能够解析和设置 Collation；而语义支持是指 TiDB 能够在比较字符串时正确地使用 Collation。
+排序规则的语法支持和语义支持受到配置项 [`new_collation_enable`](/reference/configuration/tidb-server/configuration-file.md#new_collations_enabled_on_first_bootstrap) 的影响。这里语法支持和语义支持有所区别。语法支持是指 TiDB 能够解析和设置排序规则；而语义支持是指 TiDB 能够在比较字符串时正确地使用排序规则。
 
-在 4.0 版本之前，TiDB 只提供了旧的 Collation 框架，能够在语法上支持的绝大部分 MySQL Collation，但语义上所有的 Collation 都当成 binary Collation。
+在 4.0 版本之前，TiDB 只提供了旧的排序规则框架，能够在语法上支持的绝大部分 MySQL排序规则，但语义上所有的排序规则都当成二进制排序规则。
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply reference/sql/characterset-and-collation.md:283]: LGTM

File: reference/sql/characterset-and-collation.md [hunk 10]
```diff
@@ -283,15 +278,15 @@ SET collation_connection = @@collation_database;
...
 
-4.0 版本中，TiDB 增加了新的 Collation 框架用于在语义上支持不同的 Collation，保证字符串比较时严格遵循对应的 Collation，详情请见下文。
+4.0 版本中，TiDB 增加了新的排序规则框架用于在语义上支持不同的排序规则，保证字符串比较时严格遵循对应的排序规则，详情请见下文。
 
-### 旧框架下的 Collation 支持
+### 旧框架下的排序规则支持
 
-在 4.0 版本之前，TiDB 中可以指定大部分 MySQl 中的 Collation，并把这些 Collation 按照默认 Collation 处理，即以编码字节序为字符定序。和 MySQL 不同的是，TiDB 在比较字符前按照 Collation 的 `PADDING` 属性将字符补齐空格，因此会造成以下的行为区别：
+在 4.0 版本之前，TiDB 中可以指定大部分 MySQl 中的排序规则，并把这些排序规则按照默认排序规则处理，即以编码字节序为字符定序。和 MySQL 不同的是，TiDB 在比较字符前按照 排序规则 的 `PADDING` 属性将字符补齐空格，因此会造成以下的行为区别：
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply reference/sql/characterset-and-collation.md:289]: 这里有误，应该是 “TiDB 在比较字符前按照排序规则的 `PADDING` 属性将字符末尾的空格删除”， 英文版也需要更改。

File: reference/sql/characterset-and-collation.md (hunk 1 above)
Comment (wjhuang2016) [reply reference/sql/characterset-and-collation.md:306]: LGTM

File: reference/sql/characterset-and-collation.md (hunk 2 above)
Comment (wjhuang2016) [reply reference/sql/characterset-and-collation.md:360]: LGTM

File: reference/sql/characterset-and-collation.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (yikeke) [reply reference/sql/characterset-and-collation.md:289]: @TomShawn

File: reference/sql/characterset-and-collation.md [hunk 11]
```diff
@@ -221,13 +216,13 @@ SELECT _utf8mb4'string';
 SELECT _utf8mb4'string' COLLATE utf8mb4_general_ci;
 ```
 
-规则，如下：
+规则如下：
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply reference/sql/characterset-and-collation.md:219]: 已修改

File: reference/sql/characterset-and-collation.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (TomShawn) [reply reference/sql/characterset-and-collation.md]: Updated in [#2350](https://github.com/pingcap/docs/pull/2350/commits/b9aa23a006698dbdd7b8046f746e82d136175fa8).

File: reference/sql/characterset-and-collation.md [hunk 12]
```diff
@@ -200,13 +200,13 @@ col_name {ENUM | SET} (val_list)
 
 ## 字符串的字符集和排序规则
 
-每一字符串字符文字有一个字符集和一个比较排序规则，在使用字符串时指，此选项可选，如下：
+每一个字符串都对应一个字符集和一个比较排序规则，在使用字符串时指此选项可选，如下：
```
Comment (yikeke) [inline reference/sql/characterset-and-collation.md:203]:
```suggestion
每一个字符串都对应一个字符集和一个排序规则，在使用字符串时指此选项可选，如下：
```

File: reference/sql/characterset-and-collation.md [hunk 13]
```diff
@@ -246,7 +246,7 @@ SET character_set_results = charset_name;
 SET character_set_connection = charset_name;
 ```
 
-`COLLATE` 是可选的，如果没有提供，将会用 `charset_name` 默认的排序规则。
+`COLLATE` 是可选的，如果没有提供，将会用 `charset_name` 对应的排序规则。
```
Comment (yikeke) [inline reference/sql/characterset-and-collation.md:249]:
```suggestion
`COLLATE` 是可选的，如果没有提供，将会用 `charset_name` 对应的默认排序规则。
```

File: reference/sql/characterset-and-collation.md [hunk 14]
```diff
@@ -260,15 +260,15 @@ SET character_set_results = charset_name;
...
 
-## 字符集和排序规则通用选择规则
+## 字符集和排序规则的通用选择规则
 
-* 规则 1：指定 `CHARACTER SET charset_name` 和 `COLLATE collation_name`，则直接使用 `CHARACTER SET charset_name` 和 `COLLATE collation_name`。
-* 规则 2：指定 `CHARACTER SET charset_name` 且未指定 `COLLATE collation_name`，则使用 `CHARACTER SET charset_name` 和 `CHARACTER SET charset_name` 默认的排序比较规则。
-* 规则 3：`CHARACTER SET charset_name` 和 `COLLATE collation_name` 都未指定，则使用更高优化级给出的字符集和排序比较规则。
+* 规则 1：如果指定了 `CHARACTER SET charset_name` 和 `COLLATE collation_name`，则直接使用 `CHARACTER SET charset_name` 和 `COLLATE collation_name`。
+* 规则 2：如果指定了 `CHARACTER SET charset_name` 且未指定 `COLLATE collation_name`，则使用 `CHARACTER SET charset_name` 和 `CHARACTER SET charset_name` 默认的排序规则。
+* 规则 3：如果 `CHARACTER SET charset_name` 和 `COLLATE collation_name` 都未指定，则使用更高优先级给出的字符集和排序规则。
```
Comment (yikeke) [inline reference/sql/characterset-and-collation.md:271]:
```suggestion
* 规则 3：如果 `CHARACTER SET charset_name` 和 `COLLATE collation_name` 都未指定，则使用更高优先级的字符集和排序规则。
```

File: reference/sql/characterset-and-collation.md [hunk 15]
```diff
@@ -303,7 +303,7 @@ Query OK, 1 row affected # MySQL 中，由于补齐空格比较，报错 Duplica
 
 ### 新框架下的排序规则支持
 
-TiDB 4.0 新增了完整的排序规则支持框架，从语义上支持了排序规则，并新增了配置开关 `new_collation_enabled_on_first_boostrap`，在集群初次初始化时决定是否启用新排序规则框架。在该配置开关打开之后初始化集群，可以通过 `mysql`.`tidb` 表中的 `new_collation_enabled` 变量确认新排序规则是否启用：
+TiDB 4.0 新增了完整的排序规则支持框架，从语义上支持了排序规则，并新增了配置开关 `new_collation_enabled_on_first_boostrap`，在集群初次初始化时决定是否启用新排序规则框架。在该配置开关打开之后初始化集群，可以通过 `mysql`.`tidb` 表中的 `new_collation_enabled`  变量确认是否启用新排序规则框架：
```
Comment (yikeke) [inline reference/sql/characterset-and-collation.md:306]:
```suggestion
TiDB 4.0 新增了完整的排序规则支持框架，从语义上支持了排序规则，并新增了配置开关 `new_collation_enabled_on_first_boostrap`，在集群初次初始化时决定是否启用新排序规则框架。在该配置开关打开之后初始化集群，可以通过 `mysql`.`tidb` 表中的 `new_collation_enabled` 变量确认是否启用新排序规则框架：
```

File: reference/sql/characterset-and-collation.md [hunk 16]
```diff
@@ -221,13 +216,13 @@ SELECT _utf8mb4'string';
...
 ```
 
-规则，如下：
+规则如下：
 
-* 规则1： 指定 CHARACTER SET charset_name 和 COLLATE collation_name，则直接使用 CHARACTER SET charset_name 和COLLATE collation_name。
-* 规则2:    指定 CHARACTER SET charset_name 且未指定 COLLATE collation_name，则使用 CHARACTER SET charset_name 和 CHARACTER SET charset_name 默认的排序比较规则。
-* 规则3:   CHARACTER SET charset_name 和 COLLATE collation_name 都未指定，则使用 character_set_connection 和 collation_connection 系统变量给出的字符集和比较排序规则。
+* 规则 1：如果指定了 `CHARACTER SET charset_name` 和 `COLLATE collation_name`，则直接使用 `CHARACTER SET charset_name` 和 `COLLATE collation_name`。
+* 规则 2：如果指定了 `CHARACTER SET charset_name` 且未指定 `COLLATE collation_name`，则使用 `CHARACTER SET charset_name` 和 `CHARACTER SET charset_name` 默认的排序规则。
```
Comment (yikeke) [inline reference/sql/characterset-and-collation.md:222]:
```suggestion
* 规则 2：如果指定了 `CHARACTER SET charset_name` 且未指定 `COLLATE collation_name`，则使用 `CHARACTER SET charset_name` 对应的默认排序规则。
```
有两个 `CHARACTER SET charset_name`？是写错了吗？这样改对吗 @wjhuang2016

File: reference/sql/characterset-and-collation.md [hunk 17]
```diff
@@ -265,15 +260,15 @@ SET character_set_results = charset_name;
...
 字符串 > 列 > 表 > 数据库 > 服务器 > 集群
 
-## Character Sets 和 Collation 通用选择规则
+## 字符集和排序规则的通用选择规则
 
-* 规则1： 指定 CHARACTER SET charset_name 和 COLLATE collation_name，则直接使用 CHARACTER SET charset_name 和COLLATE collation_name。
-* 规则2:    指定 CHARACTER SET charset_name 且未指定 COLLATE collation_name，则使用 CHARACTER SET charset_name 和 CHARACTER SET charset_name 默认的排序比较规则。
-* 规则3:   CHARACTER SET charset_name 和 COLLATE collation_name 都未指定，则使用更高优化级给出的字符集和排序比较规则。
+* 规则 1：如果指定了 `CHARACTER SET charset_name` 和 `COLLATE collation_name`，则直接使用 `CHARACTER SET charset_name` 和 `COLLATE collation_name`。
+* 规则 2：如果指定了 `CHARACTER SET charset_name` 且未指定 `COLLATE collation_name`，则使用 `CHARACTER SET charset_name` 和 `CHARACTER SET charset_name` 默认的排序规则。
```
Comment (yikeke) [inline reference/sql/characterset-and-collation.md:270]:
```suggestion
* 规则 2：如果指定了 `CHARACTER SET charset_name` 且未指定 `COLLATE collation_name`，则使用 `CHARACTER SET charset_name` 和 `charset_name` 对应的默认排序规则。
```

File: reference/sql/characterset-and-collation.md [hunk 18]
```diff
@@ -283,15 +278,15 @@ SET collation_connection = @@collation_database;
...
+在 4.0 版本之前，TiDB 只提供了旧的排序规则框架，能够在语法上支持的绝大部分 MySQL 排序规则，但语义上所有的排序规则都当成二进制排序规则。
 
-4.0 版本中，TiDB 增加了新的 Collation 框架用于在语义上支持不同的 Collation，保证字符串比较时严格遵循对应的 Collation，详情请见下文。
+4.0 版本中，TiDB 增加了新的排序规则框架用于在语义上支持不同的排序规则，保证字符串比较时严格遵循对应的排序规则，详情请见下文。
 
-### 旧框架下的 Collation 支持
+### 旧框架下的排序规则支持
 
-在 4.0 版本之前，TiDB 中可以指定大部分 MySQl 中的 Collation，并把这些 Collation 按照默认 Collation 处理，即以编码字节序为字符定序。和 MySQL 不同的是，TiDB 在比较字符前按照 Collation 的 `PADDING` 属性将字符补齐空格，因此会造成以下的行为区别：
+在 4.0 版本之前，TiDB 中可以指定大部分 MySQl 中的排序规则，并把这些排序规则按照默认排序规则处理，即以编码字节序为字符定序。和 MySQL 不同的是，TiDB 在比较字符前按照排序规则的 PADDING 属性将字符末尾的空格删除，因此会造成以下的行为区别：
```
Comment (yikeke) [inline reference/sql/characterset-and-collation.md:289]:
```suggestion
在 4.0 版本之前，TiDB 中可以指定大部分 MySQL 中的排序规则，并把这些排序规则按照默认排序规则处理，即以编码字节序为字符定序。和 MySQL 不同的是，TiDB 在比较字符前按照排序规则的 PADDING 属性将字符末尾的空格删除，因此会造成以下的行为区别：
```

File: reference/sql/characterset-and-collation.md [hunk 19]
```diff
@@ -265,15 +260,15 @@ SET character_set_results = charset_name;
 SET collation_connection = @@collation_database;
 ```
 
-## 集群，服务器，数据库，表，列，字符串 Character Sets 和 Collation 优化级
+## 集群、服务器、数据库、表、列、字符串的字符集和排序规则的优先级
 
 字符串 > 列 > 表 > 数据库 > 服务器 > 集群
```
Comment (yikeke) [inline reference/sql/characterset-and-collation.md:265]:
```suggestion
优先级从高到低排列顺序为：

字符串 > 列 > 表 > 数据库 > 服务器 > 集群
```

**Comment by yikeke** [comment]
This PR‘s commits need to be manually cherry-picked to release-3.1, release-3.0, and release-2.1. @TomShawn  will help open the PRs.

**Pull request merged**